### PR TITLE
Frontend preview failed display better message #19

### PR DESCRIPTION
**Pull request by longshuicy**
image
image

- export INCORE_REMOTE_HOSTNAME=https://incore-dev.ncsa.illinois.edu
- npm start
- Try DFR3 ID: 60ba57b71f2b7d4a916fae88 for example.
- Backward test: try login with commresilience and delete a item in other space; you should get the same error message as before

### CODE REVIEW

**Review by nunezco2 — APPROVED** [review]

**Comment by navarroc** [comment]
Unrelated to this PR, I get a CORS error connnecting to incore-dev. Is this new/recent? We're you able to connect?

**Comment by longshuicy** [comment]
> Unrelated to this PR, I get a CORS error connnecting to incore-dev. Is this new/recent? We're you able to connect?

It shouldn't. Similar things happened to me in the past; i have to switch to incognito mode (if you are using chrome). Could you try and see if that works?

**Comment by navarroc** [comment]
> > Unrelated to this PR, I get a CORS error connnecting to incore-dev. Is this new/recent? We're you able to connect?
> 
> It shouldn't. Similar things happened to me in the past; i have to switch to incognito mode (if you are using chrome). Could you try and see if that works?

Looks like it was because I was running with Docker. I just did npm from the console and it worked.

**Comment by navarroc** [comment]
Is it easy to know which viewer the error is coming from? Could be good to include that in the Subject, something like "Bug Report - DFR3 Viewer".

File: src/components/DFR3Viewer.jsx [hunk 1]
```diff
@@ -213,10 +215,15 @@ class DFR3Viewer extends React.Component {
 		});
 	}
 
-	// TODO set state inside component did up date is bad practice!!
+	// TODO set state inside componentDidUpdate is bad practice!!
```
Comment (navarroc) [inline src/components/DFR3Viewer.jsx:218]: Is there an issue to fix this?

**Comment by navarroc** [comment]
Overall this change looks good to me, the one question about noting which component generating the bug report could be a future improvement if we decide it's useful.

**Review by navarroc — APPROVED** [review]

**Comment by longshuicy** [comment]
> Is it easy to know which viewer the error is coming from? Could be good to include that in the Subject, something like "Bug Report - DFR3 Viewer".

Right now only DFR3 preview allows this error message report. And the wording "DFR3 Curve ID" is present in the email. I will add more information if we start to have error message feature in other viewers/places.

File: src/components/DFR3Viewer.jsx (hunk 1 above)
Comment (longshuicy) [reply src/components/DFR3Viewer.jsx:223]: I have issues to convert everything to use hook and this will go away

**Pull request merged**